### ISSUE TITLE
Globally Opt of out Tracking, but opt in for just one Entity  

### ISSUE THREAD

**Issue by geetmenon** [dotnet/efcore#26142]
This is related to ticket  #26135 . This ticket was closed prematurely before giving me the complete/correct answer. Please do not mark this ticket as a duplicate and close it without answering my question.

In the ticket #26135,  @ErikEJ  says `Use the OnConfiguringPartial method in a partial class.` .  
But there is no OnConfiguringPartial method. I am using EF Core 5.  

 I would like to set UseQueryTrackingBehavior by adding a Partial class for my DBContext. But OnConfiguring() is not a partial method of DbContext, and there is no OnConfiguringPartial  method. So how can I do it without having to subclass my DbContext (that was created using Scaffolding)?

**Comment by ErikEJ** [comment]
Sorry, got confused.

Just create the DbcontextOptions elsewhere, and use the appropriate DbContext constructor.

**Comment by geetmenon** [comment]
I don't understand your answer What do you mean by "elsewhere". Please provide the specifics (preferably with an example).

**Comment by ErikEJ** [comment]
https://docs.microsoft.com/en-us/ef/core/dbcontext-configuration/#dbcontextoptions

https://docs.microsoft.com/en-us/ef/core/dbcontext-configuration/#dbcontextoptions-versus-dbcontextoptionstcontext

**Comment by geetmenon** [comment]
These documentations are not user-friendly at all, and the answers we are looking for are buried in these long documentations ( with a  myriad of options most of the time without explaining when to use what).  Instead of pointing us to these doc links, wish you guys provide us with answers specific to our question/scenario . This would save us a lot of our time.

I googled up articles from other developers working on asp.net core projects, and this what they suggest and I plan to use:
services.AddDbContext<T>() in ConfigureServices() in Startup.cs in my asp.net core project.

For e.g. in ConfigureServices() in Startup.cs:

 services.AddDbContext<MyDbContext>(options =>
            {
                options
                .UseSqlServer(Configuration.GetConnectionString("MyDbContext"))
                .UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking); // this is the line we need to add
            }, ServiceLifetime.Transient);

Hope this will be of help to others who have the same question.

**Comment by roji** [comment]
@geetmenon I'm sorry you don't find the documentation satisfactory - we'd appreciate any concrete suggestions on how to improve them.

In any case, it seems like you have found a satisfactory solution to your problem, so I'll go ahead and close this issue. If you have any further questions don't hesitate to ask.
